### PR TITLE
Add tests for scenarios where tests are running in separate processes

### PR DESCRIPTION
**Pull request by mvorisek**
extracted from #380

runInSeparateProcess is executed very differently, so test it, fixes will be done in #380

### CODE REVIEW

File: .github/workflows/integrate.yaml [hunk 1]
```diff
@@ -548,7 +548,7 @@ jobs:
         run: "composer remove ergebnis/php-cs-fixer-config psalm/plugin-phpunit vimeo/psalm --ansi --dev --no-interaction --no-progress"
 
       - name: "Require phpunit/phpunit:^${{ matrix.phpunit-version }}"
-        run: "composer require phpunit/phpunit:^${{ matrix.phpunit-version }} --ansi --no-interaction --no-progress --update-with-all-dependencies"
+        run: "composer require phpunit/phpunit:^${{ matrix.php-version == '8.3' && matrix.phpunit-version == '9.0.0' && matrix.dependencies == 'lowest' && '9.4.3' || matrix.phpunit-version }} --ansi --no-interaction --no-progress --update-with-all-dependencies"
```
Comment (mvorisek) [inline .github/workflows/integrate.yaml:551]: https://github.com/sebastianbergmann/phpunit/pull/4486

other solution will be to patch the vendor/phpunit/phpunit

**Comment by mvorisek** [comment]
@localheinz PR should be passing, but main branch CI is broken since the last commit

**Comment by localheinz** [comment]
@mvorisek 

Can you rebase and explain what you are trying to achieve?

**Comment by mvorisek** [comment]
Rebased. This PR adds testing of PHPUnit `runInSeparateProcess` which is processed very differently [1], for example per class setup is fired per each test, thus it should contribute to per test duration. All will be fixed in #380, but let's land this testing separately.

[1] https://github.com/sebastianbergmann/phpunit/blob/10.5.3/src/Framework/TestCase.php#L639

File: .github/workflows/integrate.yaml (hunk 1 above)
Comment (localheinz) [reply .github/workflows/integrate.yaml]: See #416.

**Review by localheinz — APPROVED** [review]

**Comment by localheinz** [comment]
Thank you, @mvorisek!

**Pull request merged**